### PR TITLE
Add Environment variables required by MSBuild and node-gyp for ...

### PR DESCRIPTION
**Pull request by ahmelsayed**
…building native projects. Needed for #1076

### CODE REVIEW

**Comment by ahmelsayed** [comment]
I should add a test case as well. Will do that now

**Comment by davidebbo** [comment]
I usually use (abuse?) the command executor test for things like that to avoid adding new test. https://github.com/projectkudu/kudu/blob/master/Kudu.FunctionalTests/CommandExecutorTests.cs#L31-L45

**Comment by davidebbo** [comment]
:shipit:

**Comment by ahmelsayed** [comment]
I see I'll add it there too, but I actually meant adding a nodejs deployment test that restores the 4 packages I tested with.

**Comment by davidebbo** [comment]
Oh yes, we definitely should have that.

File: Kudu.Core/Deployment/WellKnownEnvironmentVariables.cs [hunk 1]
```diff
@@ -38,5 +38,9 @@ internal static class WellKnownEnvironmentVariables
 
         public const string GoWebConfigTemplate = "GO_WEB_CONFIG_TEMPLATE";
         public const string SelectLatestVersionCommandKey = "SELECT_LATEST_VERSION_CMD";
+
+        public const string GypMsvsVersion = "GYP_MSVS_VERSION";
+        public const string VCTargetsPath = "VCTargetsPath";
```
Comment (mathewc) [inline Kudu.Core/Deployment/WellKnownEnvironmentVariables.cs]: nit: should the variables be prefixed as VC_TargetsPath, VC_InstallDir140, to match the pattern of other vars? (e.g. GO_)

Comment (ahmelsayed) [inline Kudu.Core/Deployment/WellKnownEnvironmentVariables.cs]: These are actually consumed by MSBuild.exe so they have to match what MSBuild is expecting.

Comment (davidebbo) [inline Kudu.Core/Deployment/WellKnownEnvironmentVariables.cs]: We don't pick those. They are existing env variables that the compiler expects.

Comment (mathewc) [inline Kudu.Core/Deployment/WellKnownEnvironmentVariables.cs]: Got it.

**Pull request merged**